src/main/java/core/basesyntax/FigureSupplier.java

```diff
@@ -0,0 +1,44 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class FigureSupplier {
+    private static final int DEFAULT_NUM = 10;
+    private static final int MAX_UNITS = 20;
+    private Random random = new Random();
+    private ColorSupplier colorSupplier = new ColorSupplier();
+    private double randomSide = random.nextInt(MAX_UNITS) + 1;
+    private double randomHigh = random.nextInt(MAX_UNITS) + 1;
+    private double randomWidth = random.nextInt(MAX_UNITS) + 1;
+    private double randomFirstLeg = random.nextInt(MAX_UNITS) + 1;
+    private double randomSecondLeg = random.nextInt(MAX_UNITS) + 1;
+    private double randomRadius = random.nextInt(MAX_UNITS) + 1;
+    private double randomTrHigh = random.nextInt(MAX_UNITS) + 1;
+    private double randomTop = random.nextInt(MAX_UNITS) + 1;
+    private double randomBottom = random.nextInt(MAX_UNITS) + 1;
+    private String randomSquareColor = colorSupplier.getRandomColor();
+    private String randomRectangleColor = colorSupplier.getRandomColor();
+    private String randomTriangleColor = colorSupplier.getRandomColor();
+    private String randomCircleColor = colorSupplier.getRandomColor();
+    private String randomTrColor = colorSupplier.getRandomColor();
```

Comment:
these random parameters should be generated each time getRandomFigure() is called. So, please remove them from class fields and introduce local variables